src/solver/simulation/sim_structure_probleme_economique.h

```diff
@@ -478,11 +489,14 @@ struct PROBLEME_HEBDO
     char OptimisationAvecCoutsDeDemarrage;
     int NombreDePays;
     const char** NomsDesPays;
+    AdequacyPatchMode* AreaAdequacyPatchMode;
     int NombreDePaliersThermiques;
 
     int NombreDInterconnexions;
     int* PaysOrigineDeLInterconnexion;
     int* PaysExtremiteDeLInterconnexion;
+    AdequacyPatchMode* StartAreaAdequacyPatchType;
+    AdequacyPatchMode* EndAreaAdequacyPatchType;
     COUTS_DE_TRANSPORT** CoutDeTransport;
```

Comment:
- I think `AreaAdequacyPatchMode`, `StartAreaAdequacyPatchType` and `EndAreaAdequacyPatchType` could be defined in `AdequacyPatchParameters`. Then they would only be allocated if needed : when the adq-patch is enabled.
Otherwise memory consumption would increase even if the adq-patch is disabled, which is not desirable.

- **Origin** and **extremity** would be more appropriate than **Start** and **End** in my opinion.
- Also, instead of having two arrays for the link types (`StartAreaAdequacyPatchType` and `EndAreaAdequacyPatchType`), you could have a single array/vector containing objects of this new type
```
struct {
   AdequacyPatchMode origin;
   AdequacyPatchMode extremity;
} AdqPatchLinkType;
```

Reply:
I will work on this one.

Reply:
I abandoned this one for now.